python/src/aac/plugins/validators/_validator_result.py

```diff
@@ -1,18 +1,39 @@
-from attr import attrib, attrs, validators, Factory
+"""A module for representing validator results."""
 
 
-@attrs(slots=True, auto_attribs=True)
+from typing import Union
+from attr import Factory, attrib, attrs, validators
+
+from aac.lang.definitions.definition import Definition
+from aac.plugins.validators._validator_finding import ValidatorFinding
+from aac.plugins.validators._validator_findings import ValidatorFindings
+
+
+@attrs(slots=True)
 class ValidatorResult:
-    """Represents the validation result from a validator plugin on a single definition.
+    """
+    Represents the result of the validation of one, or more, definitions.
 
     Attributes:
-        messages (list): A dictionary of definition names as keys to lists of messages
-        is_valid (bool): A bool indicating whether the definition is valid or not (True/False)
+        findings (ValidatorFindings): A collection of findings that resulted from running validation
+                                      on the Definition.
     """
 
-    messages: list[str] = attrib(default=Factory(list), validator=validators.instance_of(list))
-    is_valid: bool = attrib(default=False, validator=validators.instance_of(bool))
+    definitions: Union[Definition, list[Definition]] = attrib(
+        default=Factory(list), validator=validators.instance_of((list, Definition))
+    )
+    findings: ValidatorFindings = attrib(
+        default=Factory(ValidatorFindings), validator=validators.instance_of(ValidatorFindings)
+    )
+
+    def is_valid(self) -> bool:
+        """Return True if there are no error messages on the validation result; False, otherwise."""
+        return len(self.findings.get_error_findings()) == 0
 
     def get_messages_as_string(self) -> str:
         """Get all of the validator result messages as a single string."""
-        return "\n".join(self.messages)
+
+        def format_message(finding: ValidatorFinding) -> str:
+            return f"\n{finding.location.validation_name}:{finding.severity.name}:{finding.location.source.uri}\n  {finding.message}"
```

Comment:
I'm torn on this error message. The ':' are good for making it machine parsable, but we may want to alter it for user readability.

My suggestion for a more human-friendly format:
```suggestion
            return f"\nValidation finding from '{finding.location.validation_name}' of level {finding.severity.name} in {finding.location.source.uri}. Message:\n {finding.message}"
```

Which will look like:
```
validate: validation_failure

Failed to validate:

Failed to validate content with errors:

Validation finding from 'Type references exist' of level ERROR in /workspace/AaC/python/model/alarm_clock/alarm_clock.yaml. Message:
 Undefined type 'Clocks' referenced: {'name': 'clock', 'type': 'Clocks'}
Validation finding from 'Type references exist' of level ERROR in /workspace/AaC/python/model/alarm_clock/alarm_clock.yaml. Message:
 Undefined type 'ClockTimer' referenced: {'name': 'timer', 'type': 'ClockTimer'}
```

Not a great format for a lot of errors, but we can make a story to improve the output -- things like grouping by file for example.